<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1273</width>
    <height>832</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Vopex 3.0 (c) Michal Vopalensky, Antonin Zajicek, ITAM CAS 2024</string>
  </property>
  <widget class="QWidget" name="mainWidget">
   <layout class="QVBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="MPLBetterCanvas" name="canvas_main" native="true"/>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Assets</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="title">
              <string>Image 1 (A)</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QComboBox" name="cb_images_a"/>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <item>
                 <widget class="QPushButton" name="pb_load_a">
                  <property name="text">
                   <string>Load</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="CustomComboBox" name="cb_save_a"/>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QSlider" name="slider_a">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_4">
             <property name="title">
              <string>Image 2 (B)</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QComboBox" name="cb_images_b"/>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QPushButton" name="pb_load_b">
                  <property name="text">
                   <string>Load</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="CustomComboBox" name="cb_save_b"/>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QSlider" name="slider_b">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_5">
             <property name="title">
              <string>Image 3 (C)</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QComboBox" name="cb_images_c"/>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QPushButton" name="pb_load_c">
                  <property name="text">
                   <string>Load</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="CustomComboBox" name="cb_save_c"/>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QSlider" name="slider_c">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_6">
             <property name="title">
              <string>Operation on images</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QComboBox" name="cb_images_r"/>
              </item>
              <item>
               <widget class="QLineEdit" name="le_operation"/>
              </item>
              <item>
               <widget class="QPushButton" name="pb_save_all_calculated">
                <property name="text">
                 <string>Save all calculated</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSlider" name="slider_d">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>350</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="title">
         <string>Figure Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QGroupBox" name="groupBox_7">
           <property name="title">
            <string>Intensity Values:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Color map:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="cb_colormaps"/>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Intensity range (image lmits xx yy)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Lower limit:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QSlider" name="slider_lower">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="dsb_lower">
                <property name="maximum">
                 <double>10000000000000.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Upper limit:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QSlider" name="slider_upper">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="dsb_upper">
                <property name="maximum">
                 <double>10000000000000.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Autorange:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <item>
               <widget class="QComboBox" name="cb_auto_range"/>
              </item>
              <item>
               <widget class="QCheckBox" name="cb_from_zoom">
                <property name="text">
                 <string>From zoom</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="pb_apply_all_range">
              <property name="text">
               <string>Apply on all</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_8">
           <property name="title">
            <string>Zoomed Area</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Current selection:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="l_curr_selection">
                <property name="text">
                 <string>N/A</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Rows range:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_14">
              <item>
               <widget class="QSpinBox" name="sb_rows_from">
                <property name="maximum">
                 <number>1000000000</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="sb_rows_to">
                <property name="maximum">
                 <number>1000000000</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Columns range:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_15">
              <item>
               <widget class="QSpinBox" name="sb_columns_from">
                <property name="maximum">
                 <number>1000000000</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="sb_columns_to">
                <property name="maximum">
                 <number>1000000000</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <item>
               <widget class="QPushButton" name="pb_unzoom">
                <property name="text">
                 <string>Entire image</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pb_apply_all_zoom">
                <property name="text">
                 <string>Apply on all</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_9">
           <property name="title">
            <string>Parameters:</string>
           </property>
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Image mean:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="l_im_mean">
              <property name="text">
               <string>N/A</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>Image sigma:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="l_im_sigma">
              <property name="text">
               <string>N/A</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Shown range mean:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLabel" name="l_range_mean">
              <property name="text">
               <string>N/A</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>Shown range sigma:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLabel" name="l_range_sigma">
              <property name="text">
               <string>N/A</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_10">
           <property name="title">
            <string>Histogram</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_9">
            <item>
             <widget class="MPLSimpleCanvas" name="canvas_histogram" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>100</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QWidget" name="widget_2" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_16">
            <item>
             <widget class="QPushButton" name="pb_save_all">
              <property name="text">
               <string>Save current image</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1273</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="a_Load_Image"/>
    <addaction name="a_Load_Images"/>
    <addaction name="a_Save_Image"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="a_Rotate_CW"/>
    <addaction name="a_Rotate_CCW"/>
    <addaction name="a_Mirror_LR"/>
    <addaction name="a_Mirror_UD"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="a_Batch_processing"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImage"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget_5">
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="tabPosition">
        <enum>QTabWidget::South</enum>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <property name="elideMode">
        <enum>Qt::ElideNone</enum>
       </property>
       <widget class="QTreeView" name="tv_open_files">
        <attribute name="title">
         <string>Open files</string>
        </attribute>
       </widget>
       <widget class="ConsoleWidget" name="console_widget">
        <attribute name="title">
         <string>IPython Console</string>
        </attribute>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="a_Load_Image">
   <property name="text">
    <string>Load Image</string>
   </property>
  </action>
  <action name="a_Load_Images">
   <property name="text">
    <string>Load Images</string>
   </property>
  </action>
  <action name="a_Save_Image">
   <property name="text">
    <string>Save Image</string>
   </property>
  </action>
  <action name="a_Rotate_CW">
   <property name="text">
    <string>Rotate CW</string>
   </property>
  </action>
  <action name="a_Rotate_CCW">
   <property name="text">
    <string>Rotate CCW</string>
   </property>
  </action>
  <action name="a_Mirror_LR">
   <property name="text">
    <string>Mirror LR</string>
   </property>
  </action>
  <action name="a_Mirror_UD">
   <property name="text">
    <string>Mirror UD</string>
   </property>
  </action>
  <action name="a_Batch_processing">
   <property name="text">
    <string>Batch processing</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MPLBetterCanvas</class>
   <extends>QWidget</extends>
   <header location="global">./ui_elements_classes/MPLBetterCanvas</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MPLSimpleCanvas</class>
   <extends>QWidget</extends>
   <header location="global">./ui_elements_classes/MPLSimpleCanvas</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ConsoleWidget</class>
   <extends>QWidget</extends>
   <header location="global">./ui_elements_classes/ConsoleWidget</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CustomComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">./ui_elements_classes/CustomComboBox</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
